lib/segment/src/data_types/named_vectors.rs

```diff
@@ -1,20 +1,21 @@
 use std::borrow::Cow;
 use std::collections::HashMap;
 
+use super::tiny_map::TinyMap;
 use super::vectors::{VectorElementType, DEFAULT_VECTOR_NAME};
 
 type CowKey<'a> = Cow<'a, str>;
 type CowValue<'a> = Cow<'a, [VectorElementType]>;
-type HashMapType<'a> = HashMap<CowKey<'a>, CowValue<'a>>;
+type HashMapType<'a> = TinyMap<CowKey<'a>, CowValue<'a>>;
```

Comment:
```suggestion
type TinyMap<'a> = tiny_map::TinyMap<CowKey<'a>, CowValue<'a>>;
```

Reply:
`HashMapType` just does not sound right, I assume you wanted a new typename but `HashMap` is already imported here, so this is my suggestion

Reply:
Thanks for reviewing, will fix